### ISSUE TITLE
look into putting source catalogs into databases

### ISSUE THREAD

**Issue by SimonKrughoff** [DarkEnergyScienceCollaboration/Twinkles#58]
It would be good to look into how to create the database tables necessary to use catalogs.  The metadata from the images will be ingested.  I think that requires ingestProcessed.py which is in:https://github.com/lsst/datarel

The ingestion of the catalog is handled by:
https://github.com/lsst/daf_ingest

**Comment by drphilmarshall** [comment]
Yes! The sooner we all become regular database users the better.

**Comment by jchiang87** [comment]
I was able to run ingestCatalog.py (from daf_ingest) on the output from the cookbook pipeline (after hacking the code to obtain the db credentials from ~/.my.cnf) via a line like this:
```
ingestCatalog.py output_data --host ki-sr01.slac.stanford.edu --port 3307 --u
ser jchiang --database jc_fermi --table Source --dstype src --id filter=r
```
This created a Source table in my KIPAC mysql tablespace that has columns corresponding to the columns in the output catalog files, e.g. output_data/src/v840-fr/R22/S11.fits:
```
mysql> describe Source;
+--------------------------------------------------------+------------+------+-----+---------+-------+
| Field                                                  | Type       | Null | Key | Default | Extra |
+--------------------------------------------------------+------------+------+-----+---------+-------+
| id                                                     | bigint(20) | NO   | PRI | NULL    |       |
| coord_ra                                               | double     | YES  |     | NULL    |       |
| coord_dec                                              | double     | YES  |     | NULL    |       |
| parent                                                 | bigint(20) | NO   |     | NULL    |       |
| flags_negative                                         | bit(1)     | NO   |     | NULL    |       |
| base_GaussianCentroid_x                                | double     | YES  |     | NULL    |       |
| base_GaussianCentroid_y                                | double     | YES  |     | NULL    |       |
| base_GaussianCentroid_flag                             | bit(1)     | NO   |     | NULL    |       |
| base_GaussianCentroid_flag_noPeak                      | bit(1)     | NO   |     | NULL    |       |
| base_NaiveCentroid_x                                   | double     | YES  |     | NULL    |       |
| base_NaiveCentroid_y                                   | double     | YES  |     | NULL    |       |
| base_NaiveCentroid_flag                                | bit(1)     | NO   |     | NULL    |       |
| base_NaiveCentroid_flag_noCounts                       | bit(1)     | NO   |     | NULL    |       |
| base_NaiveCentroid_flag_edge                           | bit(1)     | NO   |     | NULL    |       |
| base_SdssCentroid_x                                    | double     | YES  |     | NULL    |       |
| base_SdssCentroid_y                                    | double     | YES  |     | NULL    |       |
| base_SdssCentroid_xSigma                               | float      | YES  |     | NULL    |       |
| base_SdssCentroid_ySigma                               | float      | YES  |     | NULL    |       |
| base_SdssCentroid_flag                                 | bit(1)     | NO   |     | NULL    |       |
| base_SdssCentroid_flag_edge                            | bit(1)     | NO   |     | NULL    |       |
| base_SdssCentroid_flag_noSecondDerivative              | bit(1)     | NO   |     | NULL    |       |
| base_SdssCentroid_flag_almostNoSecondDerivative        | bit(1)     | NO   |     | NULL    |       |
| base_SdssCentroid_flag_notAtMaximum                    | bit(1)     | NO   |     | NULL    |       |
| base_SdssShape_xx                                      | double     | YES  |     | NULL    |       |
| base_SdssShape_yy                                      | double     | YES  |     | NULL    |       |
| base_SdssShape_xy                                      | double     | YES  |     | NULL    |       |
| base_SdssShape_xxSigma                                 | float      | YES  |     | NULL    |       |
| base_SdssShape_yySigma                                 | float      | YES  |     | NULL    |       |
| base_SdssShape_xySigma                                 | float      | YES  |     | NULL    |       |
```
Note that this doesn't really look like the Source table schema [here](https://lsst-web.ncsa.illinois.edu/schema/index.php?sVer=baseline&t=Source) so getting things into qserv-like tables must be yet another step.  Also, I don't see the visit info persisted, so sources for all 9 visits were effectively dumped into that single table.

**Comment by SimonKrughoff** [comment]
@jchiang87 Did you have to hack it to use .my.cnf because you didn't want to use the .lsst/DBauth.paf mechanism, or does that not work either?

Regarding the schema, there is no step to transform the measurement schema to that at the schema browser.  How important is that?  I think it is mostly just defining alias.

**Comment by jchiang87** [comment]
I didn't know how to use the DBauth, and saw that the .my.cnf option was available.  It was just easier to figure out how to get that working than chasing down the DBauth documentation.

Regarding the schema in the schema browser, my understanding was that those were the schemas to be used by qserv, so we wanted tables that were as close to Level 2 appearance as possible.

**Comment by jchiang87** [comment]
Over at #14 , Debbie points to https://confluence.lsstcorp.org/display/LSWUG/Process+SDSS+Stripe+82+Images where an example of ingesting forced source catalogs is given using datareal/ingestForcedSource.py, which does not appear in datarel currently.  Presumably we are to use ingestProcessed.py?

**Comment by jchiang87** [comment]
I'm going to proclaim this closed by #184.

**Comment by drphilmarshall** [comment]
Sounds good! Thanks Jim :-)
